src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -571,6 +566,52 @@ def _validate_vm_create_auth(namespace):
                 '/home/{}/.ssh/authorized_keys'.format(namespace.admin_username)
 
 
+def _validate_admin_username(username, is_linux):
+    if not username:
+        return "admin user name can not be empty"
+    # pylint: disable=line-too-long
+    pattern = (r'[\\\/"\[\]:|<>+=;,?*@#()!A-Z]+' if is_linux else r'[\\\/"\[\]:|<>+=;,?*@]+')
+    linux_err = r'admin user name cannot contain upper case character A-Z, special characters \/"[]:|<>+=;,?*@#()! or start with $ or -'
+    win_err = r'admin user name cannot contain special characters \/"[]:|<>+=;,?*@# or ends with .'
+    if re.findall(pattern, username):
+        raise CLIError(linux_err if is_linux else win_err)
+    if is_linux and re.findall(r'^[$-]+', username):
+        raise CLIError(linux_err)
+    if not is_linux and username.endswith('.'):
+        raise CLIError(win_err)
+    disallowed_user_names = [
+        "administrator", "admin", "user", "user1", "test", "user2",
+        "test1", "user3", "admin1", "1", "123", "a", "actuser", "adm",
+        "admin2", "aspnet", "backup", "console", "david", "guest", "john",
+        "owner", "root", "server", "sql", "support", "support_388945a0",
+        "sys", "test2", "test3", "user4", "user5"]
+    if username.lower() in disallowed_user_names:
+        raise CLIError('The specified admin user name is not allowed, as it uses reserved words. Try again with a different value')
+
+
+def _validate_admin_password(password, is_linux):
+    max_length = 72 if is_linux else 123
+    min_length = 12
+    if len(password) not in range(min_length, max_length + 1):
+        raise CLIError('The pssword length must be between {} and {}'.format(min_length,
+                                                                             max_length))
+    contains_lower = re.findall('[a-z]+', password)
+    contains_upper = re.findall('[A-Z]+', password)
+    contains_digit = re.findall('[0-9]+', password)
+    contains_special_char = re.findall(r'[ `~!@#$%^&*()=+_\[\]{}\|;:.\/\'\",<>?]+', password)
+    count = len([x for x in [contains_lower, contains_upper,
+                             contains_digit, contains_special_char] if x])
+    # pylint: disable=line-too-long
+    if count < 3:
+        raise CLIError('Password must have the 3 of the following: 1 lower case character, 1 upper case character, 1 number and 1 special character')
+    disallowed_passwords = [
+        "abc@123", "P@$$w0rd", "P@ssw0rd", "P@ssword123", "Pa$$word",
+        "pass@word1", "Password!", "Password1", "Password22", "iloveyou!"
+    ]
+    if password.lower() in disallowed_passwords:
+        raise CLIError('The specified password is not allowed')
```

Comment:
I'd recommend a similar error as above: that the password meets the general requirements, but it specifically disallowed.